tests/Unit/Campaigns/DataTransferObjects/CampaignGoalDataTest.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Give\Tests\Unit\Campaigns\DataTransferObjects;
+
+use Give\Campaigns\DataTransferObjects\CampaignGoalData;
+use Give\Campaigns\Models\Campaign;
+use Give\DonationForms\Models\DonationForm;
+use Give\Donations\Models\Donation;
+use Give\Donors\Models\Donor;
+use Give\Framework\Database\DB;
+use Give\Tests\TestCase;
+use Give\Tests\TestTraits\RefreshDatabase;
+use PHPUnit_Framework_MockObject_MockBuilder;
+
+/**
+ * @unreleased
+ */
+final class CampaignGoalDataTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /**
+     * @unreleased
+     */
+    public function testGetPercentageDoesNotDivideByZero()
+    {
+        $campaign = Campaign::factory()->create(['goal' => 0]);
+
+        $form = DonationForm::factory()->create();
+        DB::table('give_campaign_forms')
+            ->insert(['form_id' => $form->id, 'campaign_id' => $campaign->id]);
```

Comment:
When you create the campaign, a default form is created as well, so you don't need to create another one here. 

You can retrieve the default form id like this:

 `$campaign->defaultForm()->id;`

Reply:
Ah, interesting. That is much easier for testing.

Reply:
On second thought, I don't actually need to create the donation, since the goal progress isn't needed.

I simplified the test, see [e9a9386](https://github.com/impress-org/givewp/pull/7596/commits/e9a9386e4de8c54381ff41c6513f8d1a0ea030de).